### DISCUSSION TITLE
Why is an image imported using ctr not viewable using crictl and not recognized by k3s？

### DISCUSSION THREAD

**Discussion by Qkaiqing** [k3s-io/k3s#11159] [Q&A]
```shell
k3s --version
k3s version v1.30.5+k3s1 (9b586704)
go version go1.22.6
---
[root@VM-16-9-centos ~]# ctr -n k8s.io i import vip.tar 
unpacking ghcr.io/kube-vip/kube-vip:v0.8.4 (sha256:cb10bc6c50ff1693ed625a3b3752cd0d7edd1728bf9212fa9eceba6a7302deb2)...done
[root@VM-16-9-centos ~]# crictl images
IMAGE                                                                TAG                    IMAGE ID            SIZE
registry.cn-hangzhou.aliyuncs.com/rancher/klipper-helm               v0.9.2-build20240828   1932cb543c3e4       72.2MB
registry.cn-hangzhou.aliyuncs.com/rancher/klipper-lb                 v0.4.9                 11a5d8a9f31aa       4.99MB
registry.cn-hangzhou.aliyuncs.com/rancher/local-path-provisioner     v0.0.28                5d221316a3c61       18.4MB
registry.cn-hangzhou.aliyuncs.com/rancher/mirrored-coredns-coredns   1.11.3                 c69fa2e9cbf5f       18.6MB
registry.cn-hangzhou.aliyuncs.com/rancher/mirrored-library-traefik   2.11.8                 019c70fd40547       46.9MB
registry.cn-hangzhou.aliyuncs.com/rancher/mirrored-metrics-server    v0.7.2                 48d9cfaaf3904       19.5MB
registry.cn-hangzhou.aliyuncs.com/rancher/mirrored-pause             3.6                    6270bb605e12e       301kB
```

**Comment by brandond** [comment]
How did you create the tarball? `ctr image import` expects `docker save` format tarballs, are you perhaps providing it with something else?

Assuming its importing correctly, are you sure it's not getting pruned by the kubelet due to the disk being full? Does it show up in `ctr -n k8s.io image list`?

**Reply by Qkaiqing** [reply]
Yes, i used docker save tarball, and then I used (k3s ctr -n k8s.io i import ) to solve the problem

**Reply by brandond** [reply]
Are you running containerd standalone on the host, in addition to the containerd managed by k3s? If so that is not recommended.

**Reply by Qkaiqing** [reply]
i only used the k3s package components, other external components are not installed, before 1.30 I can use ctr -n k8s.io i import image. But now you can only use (k3s ctr -n k8s.io i import)

**Reply by brandond** [reply]
It really sounds like you're running two different container runtimes, and `k3s ctr` and `ctr` are not the same thing. 

What do you get from `ctr version` vs `k3s ctr version`?

**Reply by Qkaiqing** [reply]
My server is new and does not have any software installed. And the ctr you said he was just /usr/local/bin/K3s soft connected ctr. k3s ctr is just preceded by k3s management tools

**Reply by Qkaiqing** [reply]
```shell
[root@VM-0-14-centos ~]# ctr --version
ctr github.com/k3s-io/containerd v1.7.21-k3s2 
[root@VM-0-14-centos ~]# k3s ctr --version
ctr github.com/k3s-io/containerd v1.7.21-k3s2 
```

**Reply by brandond** [reply]
If they're the same command then they do the same things. If it says it's imported it should be imported. I suspect maybe it just got pruned by the kubelet because you were low on space? Containerd logs would say for sure.

**Reply by Qkaiqing** [reply]
I have enough disk memory,

**Comment by sebvaldez** [comment]
I had a similar issue. I decided to uninstall & reinstall WITHOUT providing `K3S_KUBECONFIG_MODE` the second time around.

```bash
# second time around I ran: 
curl -sfL https://get.k3s.io | INSTALL_K3S_EXEC="--tls-san 10.15.50.90 --disable=traefik --disable=servicelb" sh -
```

**Reply by Qkaiqing** [reply]
Ha ha, it seems like a random bug has popped up

**Comment by 7ff5fC1E397d** [comment]
Save and restore images specifying  `platform=xyz` option e.g. `linux/amd64` otherwise does not work properly 


here is how I did,

```bash
#!/bin/bash

set -euTCo pipefail
set -x

k3s crictl image ls | grep docker | perl -alne 'print "$F[0]:$F[1]"' > list.txt

while read name; do 
    k3s ctr image export ${name##*/}.tar  $name --platform=linux/amd64;
done < list.txt

### import
# ls -1 *.tar | while read name; do k3s ctr image import $name --platform=linux/amd64; done

```

**Reply by Qkaiqing** [reply]
the reason for the error is not this. I have tried it.

**Reply by Qkaiqing** [reply]
The reason for the error is not this. I have tried it.
